master/getting-started/kubernetes/installation/migration-from-flannel.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Migrating a cluster from flannel to Calico
+---
+
+# About
+
+The steps in this document explain how to migrate an existing Kubernetes cluster using Flannel networking to use Calico
+for networking.
+
+This is accomplished by installing a migration controller which performs a rolling update of the nodes in the cluster, executing
+the following steps:
+
+- Cordon and drain the node.
+- Migrate the node to use Calico.
+- Uncordon the node to allow workloads to be scheduled.
+
+At the end of you will have a fully-functional Calico cluster using VXLAN networking between pods.
+
+# Prerequisites
+
+- Migration requires a cluster using flannel for networking using the VXLAN backend.
+- Requires flannel version v0.9.1 or greater (Canal version v3.7.0 or greater).
+- Flannel must be configured to use the Kubernetes API for storing its configuration (as opposed to etcd).
+- Flannel must have been installed using a Kubernetes daemon set.
+- Your cluster must allow for modification and removal of the flannel daemon set. For example, it must not be installed using the Kubernetes add-on manager.
+
+# Procedure
+
+1. First, install Calico.
+
+   ```
+   kubectl apply -f {{site.url}}/{{page.version}}/manifests/flannel-migration/calico.yaml
+   ```
+
+   Then, install the migration controller to initiate the migration.
+
+   ```
+   kubectl apply -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
+   ```
+
+   Once applied, you will see the nodes begin to update one at a time.
+
+1. To monitor the migration, run the following command.
+
+   ```
+   kubectl get jobs -n kube-system flannel-migration
+   ```
+
+   Note that the migration controller may be rescheduled one or more times during the migration as the node hosting
+   the controller is upgraded. The installation will be complete when the ouput of the above command shows `1/1` completions. For example:
+
+   ```
+   NAME                COMPLETIONS   DURATION   AGE
+   flannel-migration   1/1           2m59s      5m9s
+   ```
+
+1. Once complete, remove the migration controller with the following command.
+
+   ```
+   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
+   ```
+# Configuration options
+
+The migration controller will auto-detect your flannel configuration, and in most cases does not require any
```

Comment:
The migration tool autodetects your Flannel configuration, and in most cases, does not require
additional configuration. If you do require special configuration, the migration tool provides the following option,
which can be set as environment variables within the pod.